### ISSUE TITLE
Input focus is not captured when use Vitest

### ISSUE THREAD

**Issue by justforuse** [vitest-dev/vitest#2013]
### Describe the bug

I am migrating from Jest to Vitest to do unit test. when I test input focus, it seems not emitted.

By the way, it work with Jest, but not Vitest

### Reproduction

StackBlitz: https://stackblitz.com/edit/vitest-dev-vitest-gndsfu?file=test%2Fbasic.test.ts,src%2Fcomponents%2FOptions.vue,package.json,vite.config.ts

In this demo, I try to trigger click and focus on input element, but only get click.

```js
await wrapper.find('input').trigger('focus');
await wrapper.find('input').trigger('click');
console.log(wrapper.emitted()); // [Object: null prototype] { click: [ [] ] }

```

```vue
<input type="text" @focus="handleFocus" @click="handleClick" />

<script>
// ...
methods: {
  handleFocus() {
    this.$emit('focus');
  },
  handleClick() {
    this.$emit('click');
  },
},
</script>
```

### System Info

```shell
System:
    OS: macOS 12.5.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 25.82 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
    npm: 7.24.2 - ~/.nvm/versions/node/v14.18.1/bin/npm
  Browsers:
    Chrome: 105.0.5195.102
    Firefox: 104.0.1
    Safari: 15.6.1
  npmPackages:
    @vitejs/plugin-vue2: ^1.1.2 => 1.1.2 
    vite: ^3.1.0 => 3.1.0 
    vitest: ^0.23.1 => 0.23.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by watonyweng** [comment]
Hi @justforuse , you may need to add a name to your `Options` component.

**Comment by justforuse** [comment]
@watonyweng  I tried add a name, but seems not solve this issue.

```vue
<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'Options',
  // xxx
});
</script>
```

**Comment by watonyweng** [comment]
@justforuse we might want to focus on the following:
1. It means your parent component will capture this method, when you call `this.$emit`.
2. Note the method signature of `emitted`, and learn more at [emitted](https://v1.test-utils.vuejs.org/api/wrapper/#emitted).

**Comment by justforuse** [comment]
@watonyweng  I just check docs again, and find the [tip]( https://v1.test-utils.vuejs.org/api/wrapper/#trigger):

>When using trigger('focus') with [jsdom v16.4.0](https://github.com/jsdom/jsdom/releases/tag/16.4.0) and above you must use the [attachTo](https://v1.test-utils.vuejs.org/api/options.html#attachto) option when mounting the component. This is because a bug fix in [jsdom v16.4.0](https://github.com/jsdom/jsdom/releases/tag/16.4.0) changed el.focus() to do nothing on elements that are disconnected from the DOM.

and test case passed. Thanks
